### PR TITLE
fix: strange menu behaviour with some child components

### PR DESCRIPTION
**Pull request by iisakkirotko**
Because `on_show_menu` hadn't been set, Python and Vue weren't in sync about the status of a menu being open.

### CODE REVIEW

File: solara/lab/components/menu.py [hunk 1]
```diff
@@ -123,13 +142,23 @@ def Page():
     ## Arguments
 
     * activator: Clicking on this element will open the menu. Accepts either a `solara.Element`, or a list of elements.
+    * open: Controls and communicates the state of the menu. If True, the menu is open. If False, the menu is closed.
     * children: List of Elements to be contained in the menu
     * style: CSS style to apply. Applied directly onto the `v-menu` component.
+    * close_on_content_click: If True, clicking on the menu contents will close the menu. If False, the menu will stay open.
```
Comment (maartenbreddels) [inline solara/lab/components/menu.py:148]: This argument is missing, I see it in #326 but you probably wanted it here.

**Review by maartenbreddels — CHANGES_REQUESTED** [review]
Nice work, just a small error I think. What happens with the ClickMenu and ContextMenu when you manually set open (or open.value) to True?

File: solara/lab/components/menu.py (hunk 1 above)
Comment (iisakkirotko) [reply solara/lab/components/menu.py:148]: My thought was that `close_on_content_click` was something to allow `InputDate` to work, and therefore shouldn't be here but only in #326. Let me know your thoughts.

Comment (maartenbreddels) [reply solara/lab/components/menu.py:148]: Ah yes, true, so we should only move the docstring there

**Comment by maartenbreddels** [comment]
Perfect 👍

**Review by maartenbreddels — APPROVED** [review]

**Pull request merged**